### DISCUSSION TITLE
3.6.11 and Google Cloud translation; still working?

### DISCUSSION THREAD

**Discussion by Sarcas666** [SubtitleEdit/subtitleedit#6773] [Q&A]
I just want to make sure it isn't something I did wrong... I'm using the 3.6.11 beta. I registered for a Google Cloud API for the Cloud Translation API. Payment method, credentials, all done, API enabled and pasted the key into SE. 

It doesn't seem to work, though. When load a srt and choosing auto-translate, I get a window with all the timestamps, but no subs. I do see the timestamps and subs in the main window. When using auto-translate with audio-to text Whisper ConstMe there seems to be translation going on, but my Google Cloud dashboard is not showing any traffic at all.

Did I miss a how-to somewhere?

thx

![afbeelding](https://user-images.githubusercontent.com/970974/227445598-1b9d5d13-a356-4adf-a889-cfa1256feba2.png)

**Comment by darnn** [comment]
Did you click "translate" (or "vertalen" in your case)?
Also, just for the record, at least for me, the Google Cloud API gives the exact same results as Google Translate itself, so you can also try it through there, if you're okay with making it a bit more tedious but not having to pay for it.

**Comment by darnn** [comment]
Also, Whisper's translate is a separate thing, it uses Whisper's own model file, the same one used for transcription, and is done locally, without connecting to the internet. It also only works even remotely well when translating into English, rather than from English into any other language. If you do indeed need to translate into English, you're usually better off transcribing the text, fixing it until you get an accurate version, and then translating it in chunks through ChatGPT ("Translate into English: [your text]"). You can do chunks of about 1,090 characters at a time, anything longer and the translation comes out truncated, or doesn't generate at all.

**Comment by niksedk** [comment]
Yes, Google Cloud translate is still working fine :)

You need to enable billing, and then create an API key, and then you need to enable the translation api:

1) Click on "Library":
![image](https://user-images.githubusercontent.com/1261331/227526959-77f32717-96c8-473d-bc5e-8417c803eb08.png)

2) Search for "Translate" and choose the "Cloud Translation API"
![image](https://user-images.githubusercontent.com/1261331/227527123-a35e856a-4e16-4915-bf91-52d979ec67f5.png)

3) Click on "Enable" and choose your billing
![image](https://user-images.githubusercontent.com/1261331/227527369-59583e45-dc55-42ea-baae-fb181f968c69.png)


And then get your API key - click on "Show key" in the "Credentials menu":
![image](https://user-images.githubusercontent.com/1261331/227528214-dede8490-ba17-4b2b-a58d-a7fa9e4138e2.png)

Insert the key into SE:
![image](https://user-images.githubusercontent.com/1261331/227528858-c5270df9-dea0-4135-989a-68f90d24a290.png)

And translate:
![image](https://user-images.githubusercontent.com/1261331/227528584-f99e8877-c78d-477c-80d5-fb0c574974ee.png)

**Reply by Sarcas666** [reply]
Thanks for your replies. I found out what I did wrong; I read in your FAQ '(hint: use main window [translate mode])', so I opened my .srt files with 'open original subtitle (translator mode)' and then it doesn't work. When I open with the regular 'open...' auto-translate works fine. I'm very keen on removing anything tedious from my life ;)

**Reply by abubrandino** [reply]
> Thanks for your replies. I found out what I did wrong; I read in your FAQ '(hint: use main window [translate mode])', so I opened my .srt files with 'open original subtitle (translator mode)' and then it doesn't work. When I open with the regular 'open...' auto-translate works fine. I'm very keen on removing anything tedious from my life ;)

Thank God for you. Exactly this fixed it. They still have the same bug even today!!!

**Comment by PreetM07** [comment]
@niksedk 
I have started facing this error.
![image](https://user-images.githubusercontent.com/63149170/232184960-1d315fd6-9508-4d17-a9b8-094da7ec2238.png)


This happens after it's done with around 300 segments or so...
It was working fine 2 days back.
I am using 3.6.7
Any idea if I am doing anything wrong?
